packages/melos/lib/src/workspace.dart

```diff
@@ -98,6 +128,9 @@ class MelosWorkspace {
   late final bool isFlutterWorkspace =
       allPackages.values.any((package) => package.isFlutterPackage);
 
+  /// Optionally specify sdk path for dart/flutter
+  late final String sdkPath;
```

Comment:
This field does not need to be `late`. 
I would prefere it if this field is nullable to signify that the sdk path is not overriden and not the empty string.

Reply:
Fixed in 79c997306f1e48af542dee1c395274ee51528330, 9364e2ed807faa6eb36779f81c3628efb0b71808 and 17b102a8a496292b1c0dfdf499494859f6f3b0d3

* Added a new function to simplify the logic here assembling the final sdk path
* Using the path package now to correctly append the paths together (to work on all platforms)
* Added tests